### DISCUSSION TITLE
Can start wayvnc server

### DISCUSSION THREAD

**Discussion by Vladimare** [any1/wayvnc#125] [General]
Hi @any1,

I messaged you in wayvnc channel a couple of days ago as vkuramshin. I work remotely and mostly on the dedicated servers but I also have my workstation in office. As I don't use it most of the time I decided to do some experiments, e.g. I installed Endeavour OS with sway. But at this point I met vnc connection issue. I found that I can use wayvnc for that. I've tried it but failed. As I wrote in IRC, first I couldn't start wayvnc and got "failed to bind address" error, then after your assumption that port 5900 could be busy (yes, qemu is using it) I specified port number as 5901 but got gnutls certificate error. You said that it's due to incorrect pem file. I'm not sure what am I doing wrong, you could see my openssl key/cert generate command, it's was taken from your GitHub readme. Anyway when I try wayvnc command remotely I get another error message: Failed to initialize wayland.
My openssl command was like:
`openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes \
	-keyout key.pem -out cert.pem -subj /CN=localhost \
	-addext subjectAltName=DNS:localhost,DNS:<hostname>,IP:<myip>`
I tried to replace all 'localhost' to my hostname but the result was the same.

I'll be happy if you help me with this.

Regards,
Vladimir Kuramshin

**Comment by any1** [comment]
What does your config file look like (with the password redacted)?

Also, make sure that you set the `WAYLAND_DISPLAY` environment variable before you run wayvnc if you're not running it from within the compositor. Most likely the correct value is `WAYLAND_DISPLAY=wayland-1`

**Comment by Vladimare** [comment]
The config is same as in the example. I just specified exact paths to key/cert files, user and password.
When I tried it on the PC the value of `WAYLAND_DISPLAY` was actually `wayland-1`. When I try wayvnc remotely it's blank.

**Comment by Vladimare** [comment]
I finally managed to launch wayvnc without errors (I exported `WAYLAND_DISPLAY=wayland-1` and specified absolute paths to key/cert files in the config) but now I cant connect to it. I try to use ultravnc viewer which worked fine with x11vnc and it shows message "No supported authentication method!"

**Reply by any1** [reply]
Yeah, the paths are relative to the location from which wayvnc is executed. Maybe they should be relative to the config file.

Ultravnc doesn't support VeNCrypt as far as I know. Tigervnc works.

**Reply by mike-yamazaki** [reply]
This seems to fail on a Raspberry Pi.

I can add the line in .bashrc. However, launching any Wayland compositor (cage or sway) fails. I take the line out and then reinitialize wayvnc inside of Sway and it works - but I have other issues. Just an FYI for anyone trying this on a Pi 4B... like me, trying the impossible
